src/postw90/get_oper.F90

```diff
@@ -1444,6 +1458,281 @@ subroutine get_SHC_R
+      scissors_shift
+    use w90_postw90_common, only: nrpts, v_matrix
+    use w90_io, only: stdout, io_error, io_stopwatch, io_file_unit, &
+      seedname
+    use w90_comms, only: on_root, comms_bcast, my_node_id
+
+    integer          :: i, j, ii, jj, m, n, a, b, nn1, nn2, ik, nb_tmp, nkp_tmp, &
+                        nntot_tmp, sIu_in, qb1, qb2, winmin_q, winmin_qb2
+    integer :: ipol
+    integer, allocatable          :: num_states(:)
+    complex(kind=dp), allocatable :: SAA_q(:, :, :, :, :)
+    complex(kind=dp), allocatable :: Ho_q_qb2(:, :, :)
+    complex(kind=dp), allocatable :: H_q_qb2(:, :)
+    real(kind=dp)                 :: c_real, c_img
+    character(len=60)             :: header
+
+    if (timing_level > 1 .and. on_root) call io_stopwatch('get_oper: get_CC_R', 1)
+
+    if (.not. allocated(BB_R)) then
+      allocate (SAA_R(num_wann, num_wann, nrpts, 3, 3))
+    else
+      if (timing_level > 1 .and. on_root) call io_stopwatch('get_oper: get_CC_R', 2)
+      return
+    end if
+
+    if (on_root) then
+
+      if (abs(scissors_shift) > 1.0e-7_dp) &
+        call io_error('Error: scissors correction not yet implemented for SAA_R')
+
+      allocate (Ho_q_qb2(num_bands, num_bands, 3))
+      allocate (H_q_qb2(num_wann, num_wann))
+      allocate (SAA_q(num_wann, num_wann, num_kpts, 3, 3))
+
+      allocate (num_states(num_kpts))
+      do ik = 1, num_kpts
+        if (have_disentangled) then
+          num_states(ik) = ndimwin(ik)
+        else
+          num_states(ik) = num_wann
+        endif
+      enddo
+
+      sIu_in = io_file_unit()
+      open (unit=sIu_in, file=trim(seedname)//".sIu", form='unformatted', &
+            status='old', action='read', err=113)
+      write (stdout, '(/a)', advance='no') &
+        ' Reading sIu overlaps from '//trim(seedname)//'.sIu in get_SAA_R: '
+      read (sIu_in, err=114, end=114) header
+      write (stdout, '(a)') trim(header)
+      read (sIu_in, err=114, end=114) nb_tmp, nkp_tmp, nntot_tmp
+      if (nb_tmp .ne. num_bands) &
+        call io_error &
+        (trim(seedname)//'.sIu has not the right number of bands')
+      if (nkp_tmp .ne. num_kpts) &
+        call io_error &
+        (trim(seedname)//'.sIu has not the right number of k-points')
+      if (nntot_tmp .ne. nntot) &
+        call io_error &
+        (trim(seedname)//'.sIu has not the right number of nearest neighbours')
+
+      SAA_q = cmplx_0
+      do ik = 1, num_kpts
+
+        call get_win_min(ik, winmin_q)
+        do nn2 = 1, nntot
+          qb2 = nnlist(ik, nn2)
+          call get_win_min(qb2, winmin_qb2)
+          do ipol = 1, 3
+            !
+            ! Read from .sIu file the matrices <u_q|s_a|u_{q+b2}>
+            ! between the original ab initio eigenstates
+            !
+            read (sIu_in, err=114, end=114) &
+              ((Ho_q_qb2(n, m, ipol), n=1, num_bands), m=1, num_bands)
+            ! pw2wannier90 is coded a bit strangely, so here we take the transpose
```

Comment:
An explanation of how the matrix elements calculated by pw2wannier90 differ from what we need here would be helpful.

Reply:
Simple, as it says, pw2wannier90 is coded with an unnecessary transposition.